node/src/accountant/db_access_objects/failed_payable_dao.rs

```diff
@@ -528,64 +566,106 @@ mod tests {
+            .build();
+        let tx4 = FailedTxBuilder::default()
+            .hash(make_tx_hash(4))
             .reason(FailureReason::NonceIssue)
+            .rechecked(false)
+            .timestamp(now - 3600) // 1 hour ago
             .build();
+        let tx5 = FailedTxBuilder::default()
+            .hash(make_tx_hash(5))
+            .reason(FailureReason::PendingTooLong)
+            .rechecked(true) // This one is rechecked
+            .timestamp(now - 3000) // 50 minutes ago
+            .build();
+
         subject
-            .insert_new_records(&vec![tx1.clone(), tx2.clone(), tx3.clone()])
+            .insert_new_records(&vec![tx1.clone(), tx2, tx3.clone(), tx4, tx5])
             .unwrap();
 
-        let result = subject.retrieve_txs(Some(ByReason(PendingTooLong)));
-
-        assert_eq!(result, vec![tx1, tx2]);
+        // Retrieve unchecked PendingTooLong transactions from the last hour
+        let result = subject.retrieve_txs(Some(FailureRetrieveCondition::UncheckedPendingTooLong(
+            3600,
+        )));
+        assert_eq!(result, vec![tx3, tx1]);
     }
 
     #[test]
-    fn update_tx_failures_works() {
+    fn update_recheck_status_works() {
         let home_dir =
-            ensure_node_home_directory_exists("failed_payable_dao", "update_tx_failures_works");
+            ensure_node_home_directory_exists("failed_payable_dao", "update_recheck_status_works");
         let wrapped_conn = DbInitializerReal::default()
             .initialize(&home_dir, DbInitializationConfig::test_default())
             .unwrap();
         let subject = FailedPayableDaoReal::new(wrapped_conn);
-        let tx1 = FailedTxBuilder::default().hash(make_tx_hash(1)).build();
-        let tx2 = FailedTxBuilder::default().hash(make_tx_hash(2)).build();
+        let tx1 = FailedTxBuilder::default()
+            .hash(make_tx_hash(1))
+            .reason(NonceIssue)
+            .rechecked(false)
+            .build();
+        let tx2 = FailedTxBuilder::default()
+            .hash(make_tx_hash(2))
+            .reason(PendingTooLong)
+            .rechecked(false)
+            .build();
+        let tx3 = FailedTxBuilder::default()
+            .hash(make_tx_hash(3))
+            .reason(PendingTooLong)
+            .rechecked(true) // already rechecked
+            .build();
+        let tx1_pre_checked_state = tx1.rechecked;
+        let tx2_pre_checked_state = tx2.rechecked;
+        let tx3_pre_checked_state = tx3.rechecked;
         subject
-            .insert_new_records(&vec![tx1.clone(), tx2.clone()])
+            .insert_new_records(&vec![tx1, tx2.clone(), tx3.clone()])
             .unwrap();
-        let hash_map = HashMap::from([(tx1.hash, PendingTooLong), (tx2.hash, NonceIssue)]);
+        let hash_set = HashSet::from([tx2.hash, tx3.hash]);
 
-        let result = subject.update_tx_failures(&hash_map);
+        let result = subject.update_recheck_status(&hash_set);
```

Comment:
As stated by the trait definition, this method is suspected of a full loss of args, because it probably won't need them anymore.